lib/components/ProfileDropdown.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useState, useRef, useEffect } from "react";
+import Link from "next/link";
+import { logout } from "@/app/login/actions";
+
+export default function ProfileDropdown() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isClicked, setIsClicked] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+        setIsClicked(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  const handleClick = () => {
+    setIsClicked(!isClicked);
+    setIsOpen(!isClicked);
+  };
+
+  const buttonStyle =
```

Comment:
Worth making a `Button` component?

Reply:
I don't think it makes sense at the moment, given one element is a Next.js Link component and the other is a regular button.